### PR TITLE
Updates to the install process

### PR DESCRIPTION
**Pull request by dwhswenson**
Just adding a little more to the docs for installing. I've been telling people that OPS is ready for them to look at; it would probably help if the docs gave accurate instructions on how to install.

Ready to merge. Only changes are in docs.

### CODE REVIEW

File: docs/getting_started.rst [hunk 1]
```diff
@@ -14,13 +14,23 @@ Install with Conda
...
+scientific python. Unlike ``easy_install`` or ``pip``, it handles binaries
+and binary dependencies, which are critical for most scientific workflows.
+If you're a ``conda`` user, you can install OpenPathSampling by adding the
+omnia channel. ::
 
-To install OpenPathSampling with conda, use the following commands ::
+To install the most recent release of OpenPathSampling with conda, use the
+following commands ::
 
   $ conda config --add channels http://conda.binstar.org/omnia
```
Comment (jchodera) [inline docs/getting_started.rst]: You just want to use `omnia` instead of `http://conda.binstar.org/omnia` now.

In fact, you may just tell them that they want to do `conda install -c omnia openpathsampling`

Comment (dwhswenson) [inline docs/getting_started.rst]: Is that `conda install -c omnia openpathsampling` as a replacement for both the commands we had previously? I'll update the docs to use that, but want to make sure they're correct (I don't know since I don't use conda, myself).

Mainly, I wanted this to have a mention of the `-dev` package, since that's the only thing that is actually currently available.

Comment (jchodera) [inline docs/getting_started.rst]: > Is that conda install -c omnia openpathsampling as a replacement for both the commands we had previously?

Yup!

>  I'll update the docs to use that, but want to make sure they're correct (I don't know since I don't use conda, myself).

It's worth testing, and won't impact your existing installation:
```bash
wget --quiet https://repo.continuum.io/miniconda/Miniconda-latest-MacOSX-x86_64.sh
bash Miniconda-latest-MacOSX-x86_64.sh -b -p miniconda
export PATH="$HOME/miniconda/bin:$PATH"
conda install -c omnia openpathsampling-dev --yes
```
Are you sure you don't want to create a non-dev version, even if it's just a preview release?

Also, what's up with no py3.x support?

Comment (franknoe) [inline docs/getting_started.rst]: Isn't the difference that in ```conda install -c omnia ...``` the 
channel is just used in this install command, rather than being added to 
your conda config? In other words, if you install by this one-liner and 
want to update subsequently, can you just do ```conda install 
openpathsampling``` or do you have to add ```-c omnia``` again?

Am 13/12/15 um 17:36 schrieb John Chodera:
>
> In docs/getting_started.rst 
> <https://github.com/choderalab/openpathsampling/pull/404#discussion_r47446925>:
>
> >
> > -To install OpenPathSampling with conda, use the following commands ::
> > +To install the most recent release of OpenPathSampling with conda, use the
> > +following commands ::
> >
> >    $ conda config --add channels http://conda.binstar.org/omnia
>
>     Is that conda install -c omnia openpathsampling as a replacement
>     for both the commands we had previously?
>
> Yup!
>
>     I'll update the docs to use that, but want to make sure they're
>     correct (I don't know since I don't use conda, myself).
>
> It's worth testing, and won't impact your existing installation:
>
> wget --quiet https://repo.continuum.io/miniconda/Miniconda-latest-MacOSX-x86_64.sh
> bash Miniconda-latest-MacOSX-x86_64.sh -b -p miniconda
> export  PATH="$HOME/miniconda/bin:$PATH"
> conda install -c omnia openpathsampling-dev --yes
>
> Are you sure you don't want to create a non-dev version, even if it's 
> just a preview release?
>
> Also, what's up with no py3.x support?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/choderalab/openpathsampling/pull/404/files#r47446925>.
>


-- 

----------------------------------------------
Prof. Dr. Frank Noe
Head of Computational Molecular Biology group
Freie Universitaet Berlin

Phone: (+49) (0)30 838 75354
Web:   research.franknoe.de
Mail:  Arnimallee 6, 14195 Berlin, Germany
----------------------------------------------

Comment (jchodera) [inline docs/getting_started.rst]: Also, you're missing some dependencies in your conda `meta.yaml`.  After installation on `osx`:
```
>>> import openpathsampling
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/openpathsampling-0.0.1a0-py2.7.egg/openpathsampling/__init__.py", line 69, in <module>
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/openpathsampling-0.0.1a0-py2.7.egg/openpathsampling/analysis/__init__.py", line 1, in <module>
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/openpathsampling-0.0.1a0-py2.7.egg/openpathsampling/analysis/histogram.py", line 3, in <module>
ImportError: No module named matplotlib.pyplot
```

Comment (jchodera) [inline docs/getting_started.rst]: To update, I think you'd have to do `conda update -c omnia openpathsampling`.

It's better in general to add the channel (`conda config --add channels omnia`), but that's a bit more advanced than the one-liner needed to install.

Comment (dwhswenson) [inline docs/getting_started.rst]: > Are you sure you don't want to create a non-dev version, even if it's just a preview release?

I have no opposition to this, but we hadn't done it. I'm just trying to make the docs reflect reality.

> Also, what's up with no py3.x support?

See #70. We decided this wasn't a priority for 1.0, but will be done soon after.

> It's better in general to add the channel (`conda config --add channels omnia`), but that's a bit more advanced than the one-liner needed to install.

I don't think a 2-line install is ***that*** much worse than a 1-line install, especially if it makes updating easier for the complete novice (which is pretty important for the pre-release versions). I'll switch it to just `--add channels omnia`.

> Also, you're missing some dependencies in your conda `meta.yaml`.

Added `matplotlib`. @jhprinz : Can we *use* the conda install for OPS on Travis instead of this separate `conda-requirements` file? That would warn us of problems like that. We would also need to find a way to distinguish between required and recommended package -- something we need to do in any case, to update the non-conda install instructions.

Comment (jhprinz) [inline docs/getting_started.rst]: The travis part is 2-fold: 
- A conda package is build using the `conda-requirements` file. 
- Then a new miniconda environment is created and the previous installed package in installed from _memory_. 

This second installation is used for testing.

Once all tests pass and only PRs on the master branch the docs are built, uploaded to amazon and the new version is uploaded to anaconda.

So we need the requirements to build the package but not afterwards. If this is still the case this should be changed.

Not sure how to have recommended packages. I presume that conda does not support this natively. We could add this to setup.py somehow?

There is also still the question if we want to add the example notebooks to the package or only put them online.

Comment (dwhswenson) [inline docs/getting_started.rst]: My point is that our Travis tests don't check whether you can actually `conda install` OPS, because they don't use the `meta.yaml` that conda uses when it installs stuff. At least, that's how I understand the fact that @jchodera got a "missing matplotlib" complaint.

The problem is that `conda-requirements` is not tied to `conda/meta.yaml`. Is there a way to fix that? If I'm understanding correctly, `conda-requirements` are not actually requirements, and not actually used by conda directly. They're "things we install using conda on Travis before we run OPS." Does conda itself ever make use of the `conda-requirements` file, or only the `meta.yaml`? Don't we want to actually install using the stuff in `meta.yaml` and make sure that we can run based on that?

My thought with "recommended packages" is that these would be other packages that aren't part of the conda `meta.yaml` but are part of the test suite. We might want to separate the testing into stages (or even separate tests) to make sure that all works correctly.

Short version:
* Currently, our tests don't check whether the conda installation will actually work. From what John had, it appears that if a user just tried to install OPS, with nothing else, the result **would not work**. We need to fix that. What's the best way to fix it?
* We will have some packages that are truly *required* for OPS to run, and some others that we want to include in testing to check that other features work. How do we manage that?

Comment (jhprinz) [inline docs/getting_started.rst]: Sorry, then I got your comment wrong. If I read the documentation of `conda build` correctly then we do not need the `conda-requirements` at all. The build will download all the files itself. Unfortunately I cannot test this on my machine becuase the require `conda-build` can only be installed in the root environment...

I will make a PR and see if that is the case. We could then simplify the procedure and remove conda-requirements
